core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerBindingService.java

```diff
@@ -113,4 +113,17 @@ public boolean exists(ConsumerBindingKey key) {
     ), null, null);
     return consumerBindingRepository.findAll(example).stream().findFirst();
   }
+
+  @PostAuthorize("returnObject.isPresent() ? hasPermission(returnObject, 'READ') : true")
+  public void findAllAndDelete(ConsumerKey key) {
+    val example = new ConsumerBinding(new ConsumerBindingKey(
+            key.getStreamDomain(),
+            key.getStreamName(),
+            key.getStreamVersion(),
+            key.getZone(),
+            null,
+            key.getName()
+    ), null, null);
+    consumerBindingRepository.findAll(example).stream().forEach(cb -> delete(cb));
```

Comment:
Looks like `findAll(T: example)` has been deprecated. Do we want to replace it's usage here with the non deprecated version?

Reply:
Not an easy way to findAll with existing methods. would require creating a filter which requires creating the query and a spec